peps/pep-0734.rst

```diff
@@ -465,39 +365,59 @@ Attributes and methods:
       (e.g. with imports) before later performing some repeated task.
 
       If there is an uncaught exception, it will be propagated into
-      the calling interpreter as a ``ExecFailure``, which
-      preserves enough information for a helpful error display.  That
-      means if the ``ExecFailure`` isn't caught then the full
-      traceback of the propagated exception, including details about
-      syntax errors, etc., will be displayed.  Having the full
-      traceback is particularly useful when debugging.
+      the calling interpreter as an ``ExecutionFailed``.  The full error
+      display of the original exception, generated relative to the
+      called interpreter, is preserved on the propagated ``ExecutionFailed``.
+      That includes the full traceback, with all the extra info like
+      syntax error details and chained exceptions.
+      If the ``ExecutionFailed`` is not caught then that full error display
+      will be shown, much like it would be if the propagated exception
+      had been raised in the main interpreter and uncaught.  Having
+      the full traceback is particularly useful when debugging.
 
       If exception propagation is not desired then an explicit
       try-except should be used around the *code* passed to
-      ``exec_sync()``.  Likewise any error handling that depends
+      ``Interpreter.exec()``.  Likewise any error handling that depends
       on specific information from the exception must use an explicit
-      try-except around the given *code*, since ``ExecFailure``
+      try-except around the given *code*, since ``ExecutionFailed``
       will not preserve that information.
 
-* ``run(code, /) -> threading.Thread``
-      Create a new thread and call ``exec_sync()`` in it.
-      Exceptions are not propagated.
+* ``call(callable, /)``
+      Call the callable object in the interpreter.
+      The return value is discarded.  If the callable raises an exception
+      then it gets propagated as an ``ExecutionFailed`` exception,
+      in the same way as ``Interpreter.exec()``.
+
+      For now only plain functions are supported and only ones that
+      take no arguments and have no cell vars.  Free globals are resolved
+      against the target interpreter's ``__main__`` module.
 
-      This is roughly equivalent to::
+      In the future, we can add support for arguments, closures,
+      and a broader variety of callables, at least partly via pickle.
+      We can also consider not discarding the return value.
+      The initial restrictions are in place to allow us to get the basic
+      functionality of the module out to users sooner.
+
+* ``call_in_thread(callable, /) -> threading.Thread``
+      Essentially, apply ``Interpreter.call()`` in a new thread.
+      Return values are discarded and exceptions are not propagated.
+
+      ``call_in_thread()`` is roughly equivalent to::
 
          def task():
-             interp.exec_sync(code)
+             interp.run(func)
```

Comment:
The description and naming suggests this:

```suggestion
             interp.call(func)
```

... but the old example suggests this:

```suggestion
             interp.exec(func)
```

Reply:
gh-3698